Storage/Processor/Processor.php

```diff
@@ -207,26 +207,39 @@ private function generateImage(Config $config, string $filename = null): string
         return $generatedImage;
     }
 
+    private function getResourceFromAsset(Config $config)
+    {
+        if ($config->getFilename()) {
+            return fopen($config->getFilename(), 'r');
+        }
 
-    public function getResource(Config $config, string $filename)
+        return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
+    }
+
+    public function getResource(Config $config, string $filename, bool $noCache = false)
     {
-        $cacheableResult = $config->cacheableResult();
         if (!$config->getStorageContext()) {
-            if ($config->getFilename()) {
-                return fopen($config->getFilename(), 'r');
-            }
+            return $this->getResourceFromAsset($config);
```

Comment:
This can still throw an error I think?

Reply:
If no file can be found the Config constructor will throw an http not found exception. So yes an error can still be thrown and the client will received a 500 error if it’s not possible to open the file